catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -698,6 +698,10 @@ public String getIdentifierQuery(String fieldName, List<? extends Serializable>
     return queryBuilder.toString();
   }
 
+  public void setAnyTextAttributes(List<String> anyTextAttributes) {
+    resolver = new DynamicSchemaResolver(anyTextAttributes, resolver.additionalFields);
```

Comment:
❗️ We shouldn't be exposing `additionalFields` like this. It breaks encapsulation. If we need to create a new DynamicSchemaResolver from a previous one then let's expose either a constructor that receives another resolver and the new text attributes or a method on the original one that creates a new one from itself with new text attributes

Reply:
❓ How do we deal with synchronization issues now that the resolver can be changed from any threads whereas before it was final?